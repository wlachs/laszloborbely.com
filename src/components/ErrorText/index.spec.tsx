import { cleanup, render, screen } from '@testing-library/react';
import { afterEach, describe, expect, it } from 'vitest';

import { ErrorText } from './';

describe('ErrorText component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should always render child elements', () => {
		const child = 'child';
		const { container } = render(<ErrorText>{child}</ErrorText>);
		expect(screen.getByText(child)).toBeDefined();
		expect(container).toMatchSnapshot();
	});

	it('should align left', () => {
		const child = 'child';
		const { container } = render(<ErrorText>{child}</ErrorText>);
		expect(container.children[0].className).not.toContain('text-center');
		expect(container).toMatchSnapshot();
	});

	it('should align center', () => {
		const child = 'child';
		const { container } = render(<ErrorText center>{child}</ErrorText>);
		expect(container.children[0].className).toContain('text-center');
		expect(container).toMatchSnapshot();
	});
});
