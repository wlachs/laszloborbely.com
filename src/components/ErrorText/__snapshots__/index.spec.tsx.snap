// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorText component > should align center 1`] = `
<div>
  <p
    class="text-center"
  >
    child
  </p>
</div>
`;

exports[`ErrorText component > should align left 1`] = `
<div>
  <p
    class=""
  >
    child
  </p>
</div>
`;

exports[`ErrorText component > should always render child elements 1`] = `
<div>
  <p
    class=""
  >
    child
  </p>
</div>
`;
